src/main/java/edu/ie3/datamodel/io/extractor/Extractor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * © 2020. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.datamodel.io.extractor;
+
+import edu.ie3.datamodel.exceptions.ExtractorException;
+import edu.ie3.datamodel.models.input.InputEntity;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * //ToDo: Class Description
+ *
+ * @version 0.1
+ * @since 31.03.20
+ */
+public class Extractor {
+
+  private final List<InputEntity> extractedEntities;
+
+  public Extractor(Nested nestedEntity) throws ExtractorException {
+    this.extractedEntities = extractElements(nestedEntity);
+  }
+
+  private List<InputEntity> extractElements(Nested nestedEntity) throws ExtractorException {
```

Comment:
```suggestion
private Set<InputEntity> extractElements(Collection<Nested> nestedEntities) throws ExtractorException {
    return Collections.unmodifiableSet(nestedEntities.stream().map(nested -> {
      if (nestedEntities instanceof Node) {
        return Collections.singletonList(((Node) nestedEntities).getNode());
      } else if (nestedEntities instanceof NodeC) {
        return Collections.singletonList(((NodeC) nestedEntities).getNodeC());
      } else if (nestedEntities instanceof Nodes) {
        return Arrays.asList(((Nodes) nestedEntities).getNodeA(), ((Nodes) nestedEntities).getNodeB());
      } else if (nestedEntities instanceof Type) {
        return Collections.singletonList(((Type) nestedEntities).getType());
      } else {
        return new ArrayList<InputEntity>();
      }
    }).flatMap(List::stream).collect(Collectors.toSet()));
  }
```

This automatically would ensure, that duplicated nodes etc. are only apparent once.